### PR TITLE
Allow admins to use pencode in faxes

### PR DESCRIPTION
**Pull request by psykzz**
## About The Pull Request

Allow admins to use pencode in faxes
## Why It's Good For The Game

pretty faxes

### CODE REVIEW

File: code/__HELPERS/pencode.dm [hunk 1]
```diff
@@ -0,0 +1,62 @@
+#define PAPER_DEFAULT_FONT "Verdana"
+#define PAPER_SIGN_FONT "Times New Roman"
+#define PAPER_CRAYON_FONT "Comic Sans MS"
+
+/proc/parse_pencode(t, obj/item/tool/pen/P, mob/user as mob, is_crayon = FALSE)
```
Comment (DominikPanic) [inline code/__HELPERS/pencode.dm]:
```suggestion
/proc/parse_pencode(t, obj/item/tool/pen/P, mob/user, is_crayon = FALSE)
```

File: code/modules/admin/verbs/faxes.dm [hunk 2]
```diff
@@ -92,7 +92,7 @@ GLOBAL_LIST_EMPTY(faxes)
 			dat += "Sender: [key_name_admin(F.sender)] [check_rights(R_ADMIN, FALSE) ? "[ADMIN_PP(F.sender)] " : ""][ADMIN_SM(F.sender)] (<a href='?src=[REF(usr.client.holder)];[HrefToken()];faxreply=[REF(F)]'>REPLY</a>) (<a href='?src=[REF(usr.client.holder)];[HrefToken()];faxview=[REF(F)]'>VIEW</a>) (<a href='?src=[REF(usr.client.holder)];[HrefToken()];faxmark=[REF(F)]'>MARK</a>)"
 		dat += "<hr>"
 
-	dat += "<a href='?src=[REF(usr.client.holder)];[HrefToken()];faxcreate=[REF(usr)]'>CREATE NEW FAX</a>"
+	dat += "<a href='?src=[REF(usr.client.holder)];[HrefToken()];faxcreate=[REF(usr)]'>Create new fax</a>"
```
Comment (DominikPanic) [inline code/modules/admin/verbs/faxes.dm]: It's big on purpose

**Comment by psykzz** [comment]
HTML is still optional, if it doesn't find the tags, it doesn't do anything.

File: code/modules/admin/verbs/faxes.dm (hunk 2 above)
Comment (psykzz) [reply code/modules/admin/verbs/faxes.dm]: Then can I bold it at least? It looks ugly.

Comment (DominikPanic) [reply code/modules/admin/verbs/faxes.dm]: It looks fine to me

**Comment by DominikPanic** [comment]
Yes but imagine if you want to add [b][/b] without having it parse. Now there is no option for that.

**Comment by psykzz** [comment]
Alright added an alert for it

**Comment by DominikPanic** [comment]
This reminds me.... Mentors shouldn't have access to HTML parsing, they should be using pencode parsing by default. Can you fix that?

File: code/modules/admin/topic.dm [hunk 3]
```diff
@@ -802,34 +802,59 @@ Status: [status ? status : "Unknown"] | Damage: [health ? health : "None"]
...
+					input = noscript(input)
+				else
+					input = sanitize(input)
+				fax_message = parse_pencode(input)
+
+				if(!fax_message)
+					return
+
+			if("Raw")
+				var/input = input("Please enter a message to send via secure connection.", "Fax Message", "") as message|null
```
Comment (DominikPanic) [inline code/modules/admin/topic.dm]: Don't use the var name input, it's confusing

File: code/modules/admin/topic.dm [hunk 4]
```diff
@@ -802,34 +802,59 @@ Status: [status ? status : "Unknown"] | Damage: [health ? health : "None"]
...
-		var/type = input("Do you want to use the template or type a custom message?", "Template") as null|anything in list("Template", "Custom")
+		var/type = input("Do you want to use the template, pencode or type a raw message?", "Fax Message") as null|anything in list("Template", "Pencode", "Raw")
 		if(!type)
 			return
 
+		var/fax_message
 		switch(type)
 			if("Template")
-				addressed_to = input("Who is it addressed to?", "Fax Message", "") as text|null
+				var/addressed_to = input("Who is it addressed to?", "Fax Message", "") as text|null
```
Comment (DominikPanic) [inline code/modules/admin/topic.dm]: You will have to sanitize every single one

File: code/modules/admin/topic.dm [hunk 5]
```diff
@@ -802,34 +802,59 @@ Status: [status ? status : "Unknown"] | Damage: [health ? health : "None"]
 			return
 
 		var/subject = input("Enter the subject line", "Fax Message", "") as text|null
-
-		var/fax_message
-		var/addressed_to
-		var/type = input("Do you want to use the template or type a custom message?", "Template") as null|anything in list("Template", "Custom")
+		var/type = input("Do you want to use the template, pencode or type a raw message?", "Fax Message") as null|anything in list("Template", "Pencode", "Raw")
```
Comment (DominikPanic) [inline code/modules/admin/topic.dm]: There are two fax functions, don't forget to modify the other one too (one is respond, one is create fax)

Comment (psykzz) [reply code/modules/admin/topic.dm]: I ended up combining both to remove some code duplication

**Pull request merged**